cis194/week2/robert-chang/src/LogAnalysis.hs

```diff
@@ -0,0 +1,39 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module LogAnalysis where
+import Data.Maybe (isJust, fromJust)
+import Text.Read ( readMaybe )
+import Log
+
+-- Exercise 1
+
+-- Helpers
+safeIntReader :: String -> Maybe Int
+safeIntReader = readMaybe
+
+getIntFromJust :: String -> Int 
+getIntFromJust s = fromJust . safeIntReader $ s 
+
+isJustValue :: String -> Bool
+isJustValue s = isJust . safeIntReader $ s
+
+parseMessage :: String -> LogMessage
+parseMessage s = case words s of 
+             ("E": level : timestamp : msg)  -> 
```

Comment:
nit: We usually have 4 space indentation and 2 space for `where` clause
```Haskell
parseMessage s = case words s of 
    ("E" ....
```

Reply:
thanks!